lib/cudnn/nnlib.jl

```diff
@@ -76,11 +104,17 @@ function conv_bias_act!(y::CuArray{T}, x::CuArray{T}, w::CuArray{T}, cdims::Dens
   # only relu and identity are supported
   if σ == NNlib.relu
     if algo < 0
-      # algo = UInt32(cudnnGetConvolutionForwardAlgorithm(fix1d(y), fix1d(x), fix1d(w), fix1d(cdims))) # will be removed in cuDNN 8
-      returnedAlgoCount, perfResults = cudnnGetConvolutionForwardAlgorithm_v7(fix1d(y), fix1d(x), fix1d(w), fix1d(cdims))
-      # returnedAlgoCount, perfResults = cudnnFindConvolutionForwardAlgorithm(fix1d(y), fix1d(x), fix1d(w), fix1d(cdims))
-      # returnedAlgoCount, perfResults = cudnnFindConvolutionForwardAlgorithmEx(fix1d(y), fix1d(x), fix1d(w), fix1d(cdims))
-      algo = UInt32(perfResults[1].algo)
+      global conv_forward
+      key = (T, size(x)[end], cdims)
+      algo = get(conv_forward, key, -1)
```

Comment:
`get!` would be nice

Reply:
in https://github.com/JuliaGPU/CUDA.jl/commit/b7d678727d87e75faa56dbe750288b5b1522c35e, I am no longer using get/push methods, but indexing with [key] instead.